lib/pdk/module/templatedir.rb

```diff
@@ -6,6 +6,11 @@ class TemplateDir
       attr_accessor :module_metadata
       attr_reader :uri
 
+      attr_reader :template_dir_type
+
+      TEMPLATE_DIR_FILESYSTEM_TYPE ||= :filesystem
```

Comment:
Move these to the module, and then no need for `||=`

Reply:
Shouldn't be needed in a class either as we're not (or shouldn't be) reloading the file and causing them to be redefined.

Reply:
Done